### PR TITLE
pre-release testing workflow for ros2

### PR DESCRIPTION
**Pull request by tylerjw**
I pushed this into the repo so we can run it on this branch to test if it works before merging.

It looks like industrial_ci supports pre-release testing for ros2: https://github.com/ros-industrial/industrial_ci/blob/master/industrial_ci/src/tests/ros_prerelease.sh#L25

### CODE REVIEW

**Comment by tylerjw** [comment]
There is not pre-release testing for galactic yet, but foxy and rolling seem to work.  https://github.com/ros-planning/moveit_msgs/runs/2831886962?check_suite_focus=true

**Comment by vatanaksoytezer** [comment]
> There is not pre-release testing for galactic yet,

Is there anything blocking that? I mean we can  add this functionality to industrial_ci I think

**Review by henningkayser — APPROVED** [review]

**Comment by tylerjw** [comment]
I reenabled the workflow dispatch test.  Merge the other commit and then that feature will work and you can merge this one and use the pre-release test on either main or ros2 branches.

File: .github/workflows/prerelease.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# This config uses industrial_ci (https://github.com/ros-industrial/industrial_ci.git).
+# For troubleshooting, see readme (https://github.com/ros-industrial/industrial_ci/blob/master/README.rst)
+
+name: pre-release
+
+on:
+  workflow_dispatch:
+  push:
```
Comment (henningkayser) [inline .github/workflows/prerelease.yaml:8]: Just for understanding... Couldn't we remove push here and get rid of the job guard?

Comment (vatanaksoytezer) [reply .github/workflows/prerelease.yaml:8]: I think so. In a related note, is there any reason not wanting prerelease tests to run with every push or nightly?

Comment (tylerjw) [reply .github/workflows/prerelease.yaml:8]: I think the reason is there is no concept of workflows that are allowed to fail with GitHub Actions.  We will want to be able to merge changes between releases that will cause the pre-release test to fail.

**Comment by tylerjw** [comment]
Now that there is a prerelease test in the default branch (master) I was able to manually trigger running the one in this branch: https://github.com/ros-planning/moveit_msgs/runs/2840796649?check_suite_focus=true

File: .github/workflows/prerelease.yaml (hunk 1 above)
Comment (tylerjw) [reply .github/workflows/prerelease.yaml]: I pushed a change removing the `push` from this.  I'm not sure what the original intent of having it here.  I mostly copied this file from the moveit repo.

**Pull request merged**